iceoryx_hoofs/include/iceoryx_hoofs/cxx/helplets.hpp

```diff
@@ -153,10 +154,10 @@ struct range
 };
 
 template <typename T>
+// AXIVION Next Construct AutosarC++19_03-A2.10.5, AutosarC++19_03-M17.0.3: The function is in the iox::cxx namespace which prevents easy misuse
 T align(const T value, const T alignment) noexcept
 {
-    T remainder = value % alignment;
-    return value + ((remainder == 0U) ? 0U : alignment - remainder);
+    return (value + (alignment - 1)) & (-alignment);
```

Comment:
We still miss documentation about the overflow and power of two constraint but OK. @FerdinandSpitzschnueffler @saif-at-github